src/dispatchers.ts

```diff
@@ -890,9 +893,17 @@ export function createOutboxDispatcher(
         );
         const outboxItems = await Promise.all(
             outbox.posts.map(async (post: Post) => {
-                const { createActivity } =
-                    await buildCreateActivityAndObjectFromPost(post, ctx);
-                return createActivity;
+                if (post.author.id === siteDefaultAccount.id) {
```

Comment:
This is gonna break if/when we add likes or any other non-author activity 🤔 

The `outboxes` table stores the type of activity right? I think we should be trying to use this data when reconstructing

I think the ideal state here would be to follow the same pattern as feeds - and have a view service which is interacting the with the outbox directly and mapping it to the DTO needed.

I think we can't easily do that as part of this PR - but it would be good to get it recorded somewhere...

Can you think of a way to expose the type from the outboxes table without restructuring everything?

Reply:
Added it here https://github.com/TryGhost/ActivityPub/pull/904/commits/b7e79a52694b581a7081b1bb795f9121e3693066